### PR TITLE
Code Quality: Remove `captionHotlinking` feature flag code 

### LINKED ISSUE #11794: Code Quality: Remove `captionHotlinking` feature flag code
## Task Description

Now that this has been enabled by default we can remove the relevant code bits.

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #11794

### CODE REVIEW

File: packages/migration/scripts/utils/updateTemplates.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ import { join, resolve } from 'path';
 /**
  * Internal dependencies
  */
-// eslint-disable-next-line import/no-unresolved
+
```
Comment (spacedmonkey) [inline packages/migration/scripts/utils/updateTemplates.js]:
```suggestion
// eslint-disable-next-line import/no-unresolved
```

File: packages/story-editor/src/components/panels/design/captions/test/captions.js [hunk 2]
```diff
@@ -74,9 +74,19 @@ describe('Panels/Captions', () => {
...
+    expect(captionButton).toBeInTheDocument();
+
+    const captionHotlinkButton = screen.queryByRole('button', {
+      name: /Link to caption file/i,
+    });
+    expect(captionHotlinkButton).toBeInTheDocument();
   });
 
-  it('should not render <Captions /> panel', () => {
+  it('should not render captions upload buton', () => {
```
Comment (spacedmonkey) [inline packages/story-editor/src/components/panels/design/captions/test/captions.js]:
```suggestion
  it('should not render captions upload button', () => {
```

**Comment by spacedmonkey** [comment]
Note that if https://github.com/GoogleForCreators/web-stories-wp/pull/11843 is merged first, this PR will need to rebased.

**Comment by timarney** [comment]
Looks like a test is failing here:
https://github.com/GoogleForCreators/web-stories-wp/runs/7116560117?check_suite_focus=true#step:7:29

**Comment by spacedmonkey** [comment]
> @spacedmonkey Looks like a test is failing here: https://github.com/GoogleForCreators/web-stories-wp/runs/7116560117?check_suite_focus=true#step:7:29

Opps, the name was input was wrong. Fixed now.

**Review by timarney — APPROVED** [review]

**Pull request merged**